app/models/agents/digest_agent.rb

```diff
@@ -40,6 +48,11 @@ def receive(incoming_events)
       incoming_events.each do |event|
         self.memory["queue"] << event.id
       end
+      if interpolated["clear_queue"].to_i > 0
+        while self.memory["queue"].length > interpolated["clear_queue"].to_i do
+          self.memory["queue"].shift
```

Comment:
You can pass the amount of elements to [`shift`](https://ruby-doc.org/core-2.2.0/Array.html#method-i-shift) which allows to remove the loop:

```ruby
if interpolated["clear_queue"].to_i > 0 && memory["queue"].length > interpolated["clear_queue"].to_i
  memory["queue"].shift(memory["queue"].length - interpolated["clear_queue"].to_i)
end
```